src/attr/_make.py

```diff
@@ -206,6 +212,7 @@ def attrib(
         converter=converter,
         metadata=metadata,
         type=type,
+        kw_only=kw_only,
```

Comment:
🤔 The current implementation raises `SyntaxError` from the generated `__init__` during class decoration. Would you prefer to raise a different exception with an explicit error message?

Reply:
I've added an explicit test of the SyntaxError behavior in 42ec86d to help clarify the discussion.

Reply:
Yeah I think I’d prefer an explicit error for that. I think it’s worth it to introduce a `PythonTooOldError("Keyword-only arguments only work on Python 3 and later.", (3, 4))` since something like this will come up more often I presume?

Reply:
It's hard to imagine a case in which calling code would catch the generated exception, so this is really a question over which error message is propagated up to the user when the system exits. I'll happily defer to your opinion, but there are a few arguments that I'd like to make explicit before adding the exception class:

1. This feature throws a relatively clear exception on python 2, and the `attrs` documentation is rather explicit about the feature being python 3+ only. From that perspective, RTFD and let's keep extra version-specific error logic out of the library.

...however...

2. `attrs` is low-level dependency for many tools. Many errors will be the result of an indirect dependency on `attrs` though some other intermediate library. Expecting potentially-less-savvy users to to read and comprehend a `SyntaxError` trace-back, then connect this to an intermediate use of the `kw_only` feature is ~~completely unreasonable~~ overly idealistic.

Reply:
I don't have a solid intuition for where this error check should occur, though my instinct is that we should find a single place to insert the check. Would you be ok with throwing from within [`_attrs_to_init_script` iff keyword-only args are being specified](https://github.com/uw-ipd/attrs/blob/42ec86d72de6b7fdfe86f4731f2dc161db79518a/src/attr/_make.py#L1473-L1477)?